src/cider/nrepl/middleware/info.clj

```diff
@@ -63,9 +75,27 @@
         (assoc :file (ns/ns-path ns)))
     meta-map))
 
+;; Similar to `print-doc` from clojure.core
+;; https://github.com/clojure/clojure/blob/master/src/clj/clojure/repl.clj#L83
+(defn- format-spec
+  [fnspec]
+  (for [role [:args :ret :fn]
+        :let [spec' (get fnspec role)]
+        :when spec']
+    (str (name role) ": " (pr-str (spec/describe spec')))))
+
+(defn maybe-add-spec
+  "If the var `nm` has a spec has associated with it, assoc that into meta-map.
+  The spec is formatted to avoid processing it in CIDER."
+  [{ns :ns nm :name :as meta-map :or {nm "" ns ""}}]
+  (if-let [fnspec (spec/get-spec (resolve-var (symbol (str ns)) (symbol nm)))]
```

Comment:
> Also, what is this doing? Isn't (resolve-var (symbol (str ns)) the same as (find-ns ns)?

We were basically reconstructing the argument `v` passed to `var-meta`, which is not included in the meta-map. It should be a var of the form, `#'some-function`. We do that with `(resolve-var ns var)` here. And `resolve-var` expects its arguments to be symbols, whereas the `ns` in this function is a `clojure.lang.Namespace` var. So we have to do`ns -> str -> symbol` conversion.

> Thirdly, what happens here if resolve-var returns nil?

`(spec/get-spec nil)` just returns nil, which is handled by the` if-let`.